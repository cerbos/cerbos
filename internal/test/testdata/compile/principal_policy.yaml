---
mainDef: "principal_policies/donald_duck_20210210.yaml"
inputDefs:
  "principal_policies/donald_duck_20210210.yaml":
    apiVersion: "api.cerbos.dev/v1"
    principalPolicy:
      principal: donald_duck
      version: "20210210"
      rules:
        - resource: leave_request
          actions:
            - action: "*"
              condition:
                match:
                  expr: "request.resource.attr.dev_record == true"
              effect: EFFECT_ALLOW

        - resource: salary_record
          actions:
            - action: "*"
              effect: EFFECT_DENY
