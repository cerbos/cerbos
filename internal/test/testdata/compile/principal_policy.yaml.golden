{
  "fqn": "cerbos.principal.donald_duck.v20210210",
  "principalPolicy": {
    "meta": {
      "fqn": "cerbos.principal.donald_duck.v20210210",
      "principal": "donald_duck",
      "version": "20210210",
      "sourceAttributes": {
        "principal.donald_duck.v20210210": {
          "attributes": {
            "source": "principal_policies/donald_duck_20210210.yaml"
          }
        }
      }
    },
    "policies": [
      {
        "resourceRules": {
          "leave_request": {
            "actionRules": [
              {
                "action": "*",
                "name": "leave_request_rule-001",
                "condition": {
                  "expr": {
                    "original": "request.resource.attr.dev_record == true",
                    "checked": {
                      "referenceMap": {
                        "1": {
                          "name": "request"
                        },
                        "5": {
                          "overloadId": [
                            "equals"
                          ]
                        }
                      },
                      "typeMap": {
                        "1": {
                          "messageType": "cerbos.engine.v1.Request"
                        },
                        "2": {
                          "messageType": "cerbos.engine.v1.Request.Resource"
                        },
                        "3": {
                          "mapType": {
                            "keyType": {
                              "primitive": "STRING"
                            },
                            "valueType": {
                              "dyn": {}
                            }
                          }
                        },
                        "4": {
                          "dyn": {}
                        },
                        "5": {
                          "primitive": "BOOL"
                        },
                        "6": {
                          "primitive": "BOOL"
                        }
                      },
                      "sourceInfo": {
                        "location": "<input>",
                        "lineOffsets": [
                          41
                        ],
                        "positions": {
                          "1": 0,
                          "2": 7,
                          "3": 16,
                          "4": 21,
                          "5": 33,
                          "6": 36
                        }
                      },
                      "expr": {
                        "id": "5",
                        "callExpr": {
                          "function": "_==_",
                          "args": [
                            {
                              "id": "4",
                              "selectExpr": {
                                "operand": {
                                  "id": "3",
                                  "selectExpr": {
                                    "operand": {
                                      "id": "2",
                                      "selectExpr": {
                                        "operand": {
                                          "id": "1",
                                          "identExpr": {
                                            "name": "request"
                                          }
                                        },
                                        "field": "resource"
                                      }
                                    },
                                    "field": "attr"
                                  }
                                },
                                "field": "dev_record"
                              }
                            },
                            {
                              "id": "6",
                              "constExpr": {
                                "boolValue": true
                              }
                            }
                          ]
                        }
                      }
                    }
                  }
                },
                "effect": "EFFECT_ALLOW",
                "output": {
                  "original": "\"donald_duck_dev_record_override:%s\".format([request.resource.attr.dev_record == true])",
                  "checked": {
                    "referenceMap": {
                      "2": {
                        "overloadId": [
                          "string_format"
                        ]
                      },
                      "4": {
                        "name": "request"
                      },
                      "8": {
                        "overloadId": [
                          "equals"
                        ]
                      }
                    },
                    "typeMap": {
                      "1": {
                        "primitive": "STRING"
                      },
                      "2": {
                        "primitive": "STRING"
                      },
                      "3": {
                        "listType": {
                          "elemType": {
                            "primitive": "BOOL"
                          }
                        }
                      },
                      "4": {
                        "messageType": "cerbos.engine.v1.Request"
                      },
                      "5": {
                        "messageType": "cerbos.engine.v1.Request.Resource"
                      },
                      "6": {
                        "mapType": {
                          "keyType": {
                            "primitive": "STRING"
                          },
                          "valueType": {
                            "dyn": {}
                          }
                        }
                      },
                      "7": {
                        "dyn": {}
                      },
                      "8": {
                        "primitive": "BOOL"
                      },
                      "9": {
                        "primitive": "BOOL"
                      }
                    },
                    "sourceInfo": {
                      "location": "<input>",
                      "lineOffsets": [
                        88
                      ],
                      "positions": {
                        "1": 0,
                        "2": 43,
                        "3": 44,
                        "4": 45,
                        "5": 52,
                        "6": 61,
                        "7": 66,
                        "8": 78,
                        "9": 81
                      }
                    },
                    "expr": {
                      "id": "2",
                      "callExpr": {
                        "target": {
                          "id": "1",
                          "constExpr": {
                            "stringValue": "donald_duck_dev_record_override:%s"
                          }
                        },
                        "function": "format",
                        "args": [
                          {
                            "id": "3",
                            "listExpr": {
                              "elements": [
                                {
                                  "id": "8",
                                  "callExpr": {
                                    "function": "_==_",
                                    "args": [
                                      {
                                        "id": "7",
                                        "selectExpr": {
                                          "operand": {
                                            "id": "6",
                                            "selectExpr": {
                                              "operand": {
                                                "id": "5",
                                                "selectExpr": {
                                                  "operand": {
                                                    "id": "4",
                                                    "identExpr": {
                                                      "name": "request"
                                                    }
                                                  },
                                                  "field": "resource"
                                                }
                                              },
                                              "field": "attr"
                                            }
                                          },
                                          "field": "dev_record"
                                        }
                                      },
                                      {
                                        "id": "9",
                                        "constExpr": {
                                          "boolValue": true
                                        }
                                      }
                                    ]
                                  }
                                }
                              ]
                            }
                          }
                        ]
                      }
                    }
                  }
                }
              }
            ]
          },
          "salary_record": {
            "actionRules": [
              {
                "action": "*",
                "name": "salary_record_rule-001",
                "effect": "EFFECT_DENY"
              }
            ]
          }
        }
      }
    ]
  },
  "compilerVersion": 1
}
